filters/include/pcl/filters/experimental/crop_box.h

```diff
@@ -0,0 +1,191 @@
+  setMin(const Eigen::Vector4f& min_pt)
+  {
+    min_pt_ = min_pt;
+  }
+
+  /** \brief Get the value of the minimum point of the box, as set by the user
+   * \return the value of the internal \a min_pt parameter.
+   */
+  inline Eigen::Vector4f
+  getMin() const
+  {
+    return min_pt_;
+  }
+
+  /** \brief Set the maximum point of the box
+   * \param[in] max_pt the maximum point of the box
+   */
+  inline void
+  setMax(const Eigen::Vector4f& max_pt)
+  {
+    max_pt_ = max_pt;
+  }
+
+  /** \brief Get the value of the maximum point of the box, as set by the user
+   * \return the value of the internal \a max_pt parameter.
+   */
+  inline Eigen::Vector4f
+  getMax() const
+  {
+    return max_pt_;
+  }
+
+  /** \brief Set a translation value for the box
+   * \param[in] translation the (tx,ty,tz) values that the box should be translated by
+   */
+  inline void
+  setTranslation(const Eigen::Vector3f& translation)
+  {
+    translation_ = translation;
+  }
+
+  /** \brief Get the value of the box translation parameter as set by the user. */
+  Eigen::Vector3f
+  getTranslation() const
+  {
+    return translation_;
+  }
+
+  /** \brief Set a rotation value for the box
+   * \param[in] rotation the (rx,ry,rz) values that the box should be rotated by
+   */
+  inline void
+  setRotation(const Eigen::Vector3f& rotation)
+  {
+    rotation_ = rotation;
+  }
+
+  /** \brief Get the value of the box rotatation parameter, as set by the user. */
+  inline Eigen::Vector3f
+  getRotation() const
+  {
+    return rotation_;
+  }
+
+  /** \brief Set a transformation that should be applied to the cloud before filtering
+   * \param[in] transform an affine transformation that needs to be applied to the
+   * cloud before filtering
+   */
+  inline void
+  setTransform(const Eigen::Affine3f& transform)
+  {
+    transform_ = transform;
+  }
+
+  /** \brief Get the value of the transformation parameter, as set by the user. */
+  inline Eigen::Affine3f
+  getTransform() const
+  {
+    return transform_;
+  }
+
+  /** \brief Sample of point indices
+   * \param[out] indices the resultant point cloud indices
+   */
+  void
+  applyFilter(Indices& indices) override
+  {
+    Eigen::Affine3f box_transform;
+    pcl::getTransformation(translation_(0),
+                           translation_(1),
+                           translation_(2),
+                           rotation_(0),
+                           rotation_(1),
+                           rotation_(2),
+                           box_transform);
+    Eigen::Affine3f pt_transform = box_transform.inverse() * transform_;
+
+    const auto lambda = [&](const PointCloud& cloud, index_t idx) {
+      const Eigen::Vector4f pt = pt_transform * cloud.at(idx).getVector4fMap();
+      return (pt.array() >= min_pt_.array()).template head<3>().all() &&
+             (pt.array() <= max_pt_.array()).template head<3>().all();
+    };
+
+    auto filter = advanced::FunctorFilter<PointT, decltype(lambda)>(
```

Comment:
Could this be static?

Reply:
Done

Reply:
I found it can't easily be static, because lambda is passed only by ctor and it avoids later initialization by ctor. It makes transformation can't be updated, as assigning a new FunctorFilter is not allowed (this is the only way I know to update a static variable)